### PR TITLE
Localstorage provider

### PR DESCRIPTION
**Pull request by DzyubSpirit**
LocalstorageProvider is ready but now webpack can't understand dependecies properly and browser build result doesn't work (test.html). See https://github.com/metarhia/globalstorage/issues/106#issuecomment-391463844.

### CODE REVIEW

**Comment by tshemsedinov** [comment]
@DzyubSpirit `Unknown authored and Unknown committed ...`
You need to fix `user.name` and `user.email` in git configuration then rewrite commits

File: data/.gs [hunk 1]
```diff
@@ -1 +1 @@
-{server:'0',count:0,size:0,next:0}
```
Comment (tshemsedinov) [inline data/.gs]: It is autogenerated file in JSON format, it will be regenerated without line end

File: lib/localstorage.provider.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+const { StorageProvider } = require('./provider');
+const { MemoryCursor } = require('./memory.cursor');
+
+function LocalstorageProvider() {}
+
+common.inherits(LocalstorageProvider, StorageProvider);
+
+LocalstorageProvider.ID_LABEL = '_LocalstorageProviderId';
+LocalstorageProvider.IDS_LABEL = '_LocalstorageProviderIds';
+LocalstorageProvider.ID_ITEM_LABEL = '_LocalstorageProvider_item_';
```
Comment (tshemsedinov) [inline lib/localstorage.provider.js]: Please add comments to describe this labels

File: package.json [hunk 3]
```diff
@@ -22,12 +22,18 @@
...
   },
   "engines": {
     "node": ">=6.0.0"
   },
+  "browser": {
+    "./lib/fs.provider.js": false,
+    "./node_modules/mkdirp": false
+  },
   "dependencies": {
+    "localStorage": "^1.0.3",
```
Comment (tshemsedinov) [inline package.json]: We need no this dependency, browser have built-in `window.localStorage`, at backend we will never use this provider

File: lib/localstorage.provider.js (hunk 2 above)
Comment (DzyubSpirit) [reply lib/localstorage.provider.js]: Thank you, I've not noticed that IDS_LABEL isn't even used.

File: package.json (hunk 3 above)
Comment (DzyubSpirit) [reply package.json]: You're right. I've moved it dev-dependencies to use it only for testing

**Comment by tshemsedinov** [comment]
We decided not to land this and suspend this to concentrate on pgsql support

**Pull request closed without merging**